inspectit-gepard-agent/src/test/java/rocks/inspectit/gepard/agent/notification/http/NotificationFactoryTest.java

```diff
@@ -25,9 +32,31 @@ void validUrlCreatesStartNotification() throws Exception {
   }
 
   @Test
-  void invalidUrlThrowsException() {
+  void invalidStartNotificationUrlThrowsException() {
     String url = "invalid url";
 
     assertThrows(URISyntaxException.class, () -> NotificationFactory.createStartNotification(url));
   }
+
+  @Test
+  void validUrlCreatesShutdownNotification() throws Exception {
+    String baseUrl = "http://localhost:8080";
+    String url = "http://localhost:8080/connections";
+    String contentType = "application/json";
+    String body = mapper.writeValueAsString(ShutdownNotification.INSTANCE);
+
+    SimpleHttpRequest request = NotificationFactory.createShutdownNotification(baseUrl);
+
+    assertEquals(url, request.getUri().toString());
+    assertEquals(contentType, request.getHeader("content-type").getValue());
+    assertEquals(body, request.getBodyText());
+  }
+
+  @Test
+  void invalidShutdownUrlThrowsException() {
+    String url = "invalid url";
+
+    assertThrows(
+        URISyntaxException.class, () -> NotificationFactory.createShutdownNotification(url));
+  }
```

Comment:
What happens, if an exception is thrown at shutdown?
Maybe not the right class to test this, but might be interesting.

Reply:
An not caught exception should always kill it's thread :)